tests/ui/abi/statics/static-mut-foreign.stderr

```diff
@@ -1,31 +1,101 @@
+  --> $DIR/static-mut-foreign.rs:28:16
+   |
+LL |     assert_eq!(rust_dbg_static_mut, 5);
+   |                ^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
+   |
+   = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
+   = note: this will be a hard error in the 2024 edition
+   = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+help: use `addr_of!` instead to create a raw pointer
+   |
+LL |     assert_eq!(addr_of!(rust_dbg_static_mut), 5);
+   |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+warning: creating a shared reference to mutable static is discouraged
+  --> $DIR/static-mut-foreign.rs:31:16
+   |
+LL |     assert_eq!(rust_dbg_static_mut, 15);
+   |                ^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
+   |
+   = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
+   = note: this will be a hard error in the 2024 edition
+   = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+help: use `addr_of!` instead to create a raw pointer
+   |
+LL |     assert_eq!(addr_of!(rust_dbg_static_mut), 15);
+   |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+warning: creating a shared reference to mutable static is discouraged
+  --> $DIR/static-mut-foreign.rs:34:16
+   |
+LL |     assert_eq!(rust_dbg_static_mut, -3);
+   |                ^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
+   |
+   = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
+   = note: this will be a hard error in the 2024 edition
+   = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+help: use `addr_of!` instead to create a raw pointer
+   |
+LL |     assert_eq!(addr_of!(rust_dbg_static_mut), -3);
+   |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+warning: creating a shared reference to mutable static is discouraged
+  --> $DIR/static-mut-foreign.rs:36:19
+   |
+LL |     static_bound(&rust_dbg_static_mut);
+   |                   ^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
+   |
+   = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
+   = note: this will be a hard error in the 2024 edition
+   = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+help: use `addr_of!` instead to create a raw pointer
+   |
+LL |     static_bound(&addr_of!(rust_dbg_static_mut));
```

Comment:
There is something weird going on here (and in other tests too). It's suggesting `&addr_of!(FOO)` or `&mut addr_of!(FOO)` when it should just be `addr_of!(FOO)`.